src/MSALWrapper/AuthFlow/CachedAuth.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Authentication.MSALWrapper.AuthFlow
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+
+    using Microsoft.Extensions.Logging;
+    using Microsoft.Identity.Client;
+
+    /// <summary>
+    /// This class provides an abstraction over how to use the MSAL Cached Auth (known as GetTokenSilent).
+    /// </summary>
+    public static class CachedAuth
+    {
+        /// <summary>
+        /// Try to get a token silently.
+        /// </summary>
+        /// <param name="logger">An <see cref="ILogger"/>.</param>
+        /// <param name="cachedAuthTimeout">The timeout for getting a cached token.</param>
+        /// <param name="scopes">Auth Scopes.</param>
+        /// <param name="account"><see cref="IAccount"/> object to use or null if no account has been found.</param>
+        /// <param name="pcaWrapper">An <see cref="IPCAWrapper"/>.</param>
+        /// <param name="errors">List of errors to report any Exceptions into.</param>
+        /// <returns>A <see cref="Task{TResult}"/> or null if no cached token was acquired.</returns>
+        public static async Task<TokenResult> TryCachedAuthAsync(ILogger logger, TimeSpan cachedAuthTimeout, IEnumerable<string> scopes, IAccount account, IPCAWrapper pcaWrapper, IList<Exception> errors)

```

Comment:
I feel this method has too many arguments. I don't know if it fits the scope of PR but we should look to simplify

Reply:
That's a great question. I built this fairly mechanically to extract the code as it was. Given this is always trying to use the cache, we could at the very least not require a cached auth timeout, and instead define that constant on this class. The rest would require a lot more thought about how to get that data over here.

Reply:
A potentially easy thing to reach for would have been putting this method on the AuthFlowBase class, to be shared by all implemtnors, and move the logger, and errors into it. But, inheritance used for behavior sharing (rather than enforcement) can lead to the wrong abstraction.

This class sets us up nicely in the future to actually move the cached auth into it's own AuthFlow.

Reply:
I agree. I am fine with removing cached auth timeout parameter for now